src/extension/tools/node/getErrorsTool.tsx

```diff
@@ -140,6 +144,21 @@ class GetErrorsTool extends Disposable implements ICopilotTool<IGetErrorsParams>
 		return uris.map(formatUriForFileWidget).join(', ');
 	}
 
+	private getNotebookCellDiagnostics(uri: URI) {
+		const notebook = this.workspaceService.notebookDocuments
```

Comment:
I would suggest using `findNotebook` function from `notebooks.ts` (that does what you're doing here and you also don't need to do the `uri.toString`)